components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/CertificateMgtDAO.java

```diff
@@ -114,26 +114,35 @@ public boolean isTableExists() throws CertificateManagementException {
         String getCertsQuery = SQLConstants.CertificateConstants.GET_CERTIFICATES;
 
         try {
-            connection = APIMgtDBUtil.getConnection();
-            connection.commit();
-            preparedStatement = connection.prepareStatement(getCertsQuery);
-            preparedStatement.setInt(1, tenantId);
-            resultSet = preparedStatement.executeQuery();
+            try {
+                connection = APIMgtDBUtil.getConnection();
+                initialAutoCommit = connection.getAutoCommit();
+                connection.setAutoCommit(false);
+                connection.commit();
+                preparedStatement = connection.prepareStatement(getCertsQuery);
+                preparedStatement.setInt(1, tenantId);
+                resultSet = preparedStatement.executeQuery();
 
-            while (resultSet.next()) {
-                certificateMetadataDTO = new CertificateMetadataDTO();
-                certificateMetadataDTO.setAlias(resultSet.getString("ALIAS"));
-                certificateMetadataDTO.setEndpoint(resultSet.getString("END_POINT"));
-                certificates.add(certificateMetadataDTO);
+                while (resultSet.next()) {
+                    certificateMetadataDTO = new CertificateMetadataDTO();
+                    certificateMetadataDTO.setAlias(resultSet.getString("ALIAS"));
+                    certificateMetadataDTO.setEndpoint(resultSet.getString("END_POINT"));
+                    certificates.add(certificateMetadataDTO);
+                }
+            } catch (SQLException e) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Error occurred while retrieving certificate metadata. ", e);
+                }
+                handleException("Error while retrieving certificates.", e);
+            } finally {
+                APIMgtDBUtil.closeStatement(preparedStatement);
+                APIMgtDBUtil.closeAllConnections(preparedStatement, connection, resultSet);
+                connection.setAutoCommit(initialAutoCommit);
             }
         } catch (SQLException e) {
             if (log.isDebugEnabled()) {
-                log.debug("Error occurred while retrieving certificate metadata. ", e);
+                log.debug("Error occurred in finally block while retrieving certificate metadata. ", e);
             }
-            handleException("Error while retrieving certificates.", e);
```

Comment:
We are removing this line? Which means, the exception is swallowed if debug is not enabled?

Reply:
fixed with https://github.com/wso2/carbon-apimgt/commit/3c8d1e61dc642ebe8179c483156798312a03382b